core/trino-main/src/test/java/io/trino/type/TestUuidType.java

```diff
@@ -61,4 +73,55 @@ public void testDisplayName()
     {
         assertEquals(UUID.getDisplayName(), "uuid");
     }
+
+    @Test
+    public void testJavaUuidToTrinoUuid()
+    {
+        assertThat(javaUuidToTrinoUuid(java.util.UUID.fromString("00000000-0000-0000-0000-000000000001")))
+                .isEqualTo(castFromVarcharToUuid(utf8Slice("00000000-0000-0000-0000-000000000001")));
+
+        assertThat(javaUuidToTrinoUuid(java.util.UUID.fromString("f79c3e09-677c-4bbd-a479-3f349cb785e7")))
+                .isEqualTo(castFromVarcharToUuid(utf8Slice("f79c3e09-677c-4bbd-a479-3f349cb785e7")));
+    }
+
+    @Test
+    public void testOrdering()
+            throws Throwable
+    {
+        String lowerAsString = "406caec7-68b9-4778-81b2-a12ece70c8b1";
+        String higherAsString = "f79c3e09-677c-4bbd-a479-3f349cb785e7";
+        java.util.UUID lower = java.util.UUID.fromString(lowerAsString);
+        java.util.UUID higher = java.util.UUID.fromString(higherAsString);
+        // Java UUID's comparison is not consitent with RFC 4122, see https://bugs.openjdk.org/browse/JDK-7025832
+        assertThat(higher).isLessThan(lower);
+
+        Slice lowerSlice = javaUuidToTrinoUuid(lower);
+        Slice higherSlice = javaUuidToTrinoUuid(higher);
+
+        assertThat(Ints.lexicographicalComparator().compare(toUnsignedBytes(lowerSlice), toUnsignedBytes(higherSlice)))
```

Comment:
`Slice` already compares lexicographically based on the bit representation, so the conversion to bytes and unsignedBytes is unnecessary

Reply:
Does slice compare bytes as unsigned?

Reply:
it does, and it's documented.